LanguageFeatures/Augmentation-libraries/augmenting_types_A06_t07.dart

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2024, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A class, enum, extension type, or mixin augmentation may specify
+/// extends, implements, on, and with clauses (when generally supported). The
+/// types in these clauses are appended to the original declarations clauses of
+/// the same kind, and if that clause did not exist previously then it is added
+/// with the new types. All regular rules apply after this appending process, so
+/// you cannot have multiple extends on a class, or an on clause on an enum, etc
+///
+/// @description Checks that a mixin augment may specify an `on` clause
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=macros
+
+import augment 'augmenting_types_A06_t07_lib.dart';
+
+class A {}
+
+class C implements A {}
+
+mixin M on A {}
+
+class MA = C with M;
```

Comment:
I think it could be useful to ensure that `M` is `on A` and also `on C`. As it is tested currently, we could have forgotten all about `A` and just turned `M` into `mixin M on C {}`.

We could use the following to make sure it is actually `mixin M on A, C {}`:

```dart
class A {}
class C {}
class D implements A, C {}

mixin M on A {} // Augmented to `on A, C`.

class MA = D with M; // OK
class BadMA1 = A with M; // Error.
class BadMA2 = C with M; // Error.
```

This would probably be done in two tests such that one can be compiled and executed, and the other one just checks that we get the expect compile-time errors.